packages/ui/src/components/cell/cell.native.tsx

```diff
@@ -1,108 +1,84 @@
-import { Switch } from 'react-native';
+import { ReactNode } from 'react';
 
 import { ResponsiveValue, useTheme } from '@shopify/restyle';
 
-import { Box, ChevronRightIcon, IconProps, Text, Theme, TouchableOpacity } from '../../../native';
-import { TouchableOpacityProps } from '../button/touchable-opacity.native';
+import {
+  Avatar,
+  ChevronRightIcon,
+  Flag,
+  IconProps,
+  ItemLayout,
+  RadioButton,
+  Theme,
+  Switch as UISwitch,
+} from '../../../native';
+import { Pressable, PressableProps } from '../button/pressable.native';
+import { RadioButtonProps } from '../radio-button/radio-button.native';
+import { SwitchProps } from '../switch/switch.native';
 
 type RegularCellVariant = 'active' | 'inactive' | 'critical';
-type SwitchCellVariant = 'switch';
-type CellVariant = RegularCellVariant | SwitchCellVariant;
 
 function getIconColor(
-  variant: CellVariant
+  variant: RegularCellVariant
 ): ResponsiveValue<keyof Theme['colors'], Theme['breakpoints']> {
   switch (variant) {
     case 'active':
-    case 'switch':
       return 'ink.text-primary';
     case 'inactive':
       return 'ink.text-subdued';
     case 'critical':
       return 'red.action-primary-default';
   }
 }
-
-function getIconBackgroundColor(
-  variant: CellVariant
-): ResponsiveValue<keyof Theme['colors'], Theme['breakpoints']> {
-  switch (variant) {
-    case 'active':
-    case 'switch':
-      return 'ink.background-secondary';
-    case 'inactive':
-      return 'ink.background-secondary';
-    case 'critical':
-      return 'red.background-primary';
-  }
+interface CellProps extends PressableProps<Theme> {
+  caption?: string;
+  Icon?: React.FC<IconProps>;
+  title: string;
+  children: NonNullable<ReactNode>;
+  variant: RegularCellVariant;
 }
 
-interface SwitchCell {
-  variant: SwitchCellVariant;
-  switchValue: boolean;
-  toggleSwitchValue(): void;
-}
+export function Cell({ caption, Icon, title, children, variant, ...props }: CellProps) {
+  const theme = useTheme<Theme>();
+
+  const itemLayout = <ItemLayout actionIcon={children} captionLeft={caption} titleLeft={title} />;
 
-interface RegularCell {
-  variant?: RegularCellVariant;
+  const content = Icon ? (
+    <Flag
+      img={
+        <Avatar>
+          <Icon color={theme.colors[getIconColor(variant)]} />
+        </Avatar>
+      }
+    >
+      {itemLayout}
+    </Flag>
+  ) : (
+    itemLayout
+  );
+
+  return (
+    <Pressable flexDirection="row" {...props}>
+      {content}
+    </Pressable>
+  );
 }
 
-interface BaseCell {
-  title: string;
-  subtitle?: string;
-  onPress?(): unknown;
-  Icon?: React.FC<IconProps>;
+function Press() {
```

Comment:
Wonder if there is a better name here? All of them are using `Pressable` ...if specific to using the chevron icon, maybe just `Chevron`?